File: src/main/java/kr/mjc/rachel/basicsyntax/SafeCalls.java
package kr.mjc.rachel.basicsyntax;

public class SafeCalls {
  public static void main(String[] args) {
    String a = "abc";
    String b = "abc";
    b = null;
    System.out.println(b);

    System.out.println("a.length = " + a.length());
    System.out.println("b.length = " + ((b != null) ? b.length() : null));
  }
}

File: src/main/kotlin/kr/mjc/jacob/basicsyntax/SafeCalls.kt
/*
 * https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-nullable-types
 */
package kr.mjc.jacob.basicsyntax

/**
 * Safe calls
 */
fun main() {
  var a: String = "abc" // a는 null이 될 수 없다.
  // a = null // compilation error

  var b: String? = "abc" // b는 null이 될 수 있다. null이 가능한 변수에 ? 붙임
  b = null // ok
  println(b)

  println("a.length = ${a.length}")
  println("b.length = ${b?.length}") // if (b!=null) b.length else null
}